File: WebBEArtGallery/Models/Dtos/GalleryDTO.cs
﻿using System;
using System.Collections.Generic;

namespace WebBEArtGallery.Models.Dtos
{
    public class GalleryDTO
    {
        public int Id { get; set; } 
        public string GalleryName { get; set; }
        public string GalleryAddress { get; set; }
        public DateTime? GalleryDateOfEstablishment { get; set; }
        public string GalleryDescription { get; set; }

        public List<ArtworkDTO> Exhibited_Artworks { get; set; }
    }
}

File: art-gallery-app/src/models/galleryDTO.ts
import { ArtworkDTO } from "./artworkDTO";

export interface GalleryDTO {
    Id: number;
    GalleryName: string;
    GalleryAddress: string;
    GalleryDateOfEstablishment?: Date; // Optional
    GalleryDescription?: string; // Optional
  
    // This could be a list of artworks or could be omitted if not needed
    Exhibited_Artworks?: ArtworkDTO[]; // Optional
  }
  